File: src/1974.minimum-time-to-type-word-using-special-typewriter.2088/solution.js
/**
 * @param {string} word
 * @return {number}
 */
var minTimeToType = function (word) {
    let prevCode = 'a'.charCodeAt(0);
    let result = word.length;
    for (let i = 0; i < word.length; i++) {
        const code = word.charCodeAt(i);
        const delta = Math.abs(code - prevCode);
        result += Math.min(delta, 26 - delta);
        prevCode = code;
    }
    return result;
};

File: src/1974.minimum-time-to-type-word-using-special-typewriter.2088/solution.java
class Solution {
    public int minTimeToType(String word) {
        char last = 'a';
        char[] chars = word.toCharArray();
        int result = 0;
        for(char c : chars){
            int diff = Math.abs(c-last);
            result += Math.min(diff,26-diff);
            last = c;
        }
        return result+chars.length;
    }
}